vars/stepResult.groovy

```diff
@@ -20,114 +20,113 @@ def call(Map config= [:]) {
-            config['result'] == "UNSTABLE" ||
-            config['result'] == "FAILURE") {
-             def comment_url = env.BUILD_URL + "display/redirect"
-
-            if (env.CHANGE_ID) {
-                if (log_url) {
-                    comment_url = log_url
-                }
-
-                pullRequest.comment("Test stage ${config.name}" +
-                                    " completed with status " +
-                                    "${config.result}" +
-                                    ".  " + comment_url)
-            }
-        }
-
-        def result = config['result']
-        switch(config['result']) {
-            case "UNSTABLE":
-                result = "FAILURE"
-                break
-            case "FAILURE":
-                result = "ERROR"
-                break
-        }
-        Map param =[:]
-        param['description'] = config['name']
-        if (config['context'].contains('/')) {
-            // Allow migration to common context value for scmNotify and
-            // stepResult
-           param['context'] = config['context']
-        } else {
-           param['context'] = config['context'] + "/" + config['name']
-        }
-        if (log_url) {
-           param['targetURL'] = log_url
-        }
-        param['status'] = result
-
-        scmNotify param
+  Map param =[:]
+  param['description'] = config['name']
+  if (config['context'].contains('/')) {
+    // Allow migration to common context value for scmNotify and
+    // stepResult
+    param['context'] = config['context']
+  } else {
+    param['context'] = config['context'] + "/" + config['name']
+  }
+  def flow_name = config.get('flow_name', env.STAGE_NAME)
+
+  def log_url = null
+
+  if (env.DAOS_JENKINS_NOTIFY_STATUS == null) {
+    println "Jenkins not configured to notify github of builds."
+    return
+  }
+
+  if (config['junit_files'] && config['result'] != 'FAILURE') {
+    log_url = env.JOB_URL - ~/job\/[^\/]*\/$/ +
+              "/view/change-requests/job/${env.BRANCH_NAME}/" +
+              "${env.BUILD_ID}/testReport/(root)/"
+  } else {
+
+    def h = new com.intel.doGetHttpRequest()
+    resp = h.doGetHttpRequest(env.JOB_URL - ~/\/job\/[^\/]*\/$/ +
+           "/view/change-requests/job/" +
+           env.BRANCH_NAME.replaceAll('/', '%252F') +
+           "/${env.BUILD_ID}/wfapi/describe");
+
+    def job = readJSON text: resp
+    assert job instanceof Map
+
+    def stage
+    for (s in job['stages']) {
+      if (s['name'] == env.STAGE_NAME) {
+        stage = s
+        break
+      }
+    }
+    resp = h.doGetHttpRequest("${env.JENKINS_URL}" +
+           stage['_links']['self']['href'])
+
+    stage = readJSON text: resp
+    assert stage instanceof Map
+
+    def stageFlowNode = null
+    for (s in stage['stageFlowNodes']) {
+      if (s['name'] == flow_name) {
+        stageFlowNode = s
+        break
+      }
+    }
+    if (!stageFlowNode) {
+      echo "No step named \"" + env.STAGE_NAME +
```

Comment:
Shouldn't this now be:
```suggestion
      echo "No step named \"" + flow_name +
```

And that fact that we are looking for a step name and reporting it's being missing really seems to indicate to me that allowing the setting of a flow_name just doesn't have any use.  If some step were to override the flow name, it would be unclear why a step of some different name were being looked for.